### PR TITLE
[doc] Remove develop.rst

### PR DESCRIPTION
**Pull request by stephenfin**
This contained a list of extensions, along with general contributing information and pointers on how to develop your own extensions. The list of extensions was very old, referenced the dead BitBucket repo and was really not very helpful, while the contributing and development information was just pointers to other, better documentation. Remove the document wholesale and update any pointers we have to it to other, better docs.

### CODE REVIEW

**Review by tk0miya — APPROVED** [review]
LGTM. BTW, PyPI provides a classifier for sphinx extensions. So it would be nice if we have a reference for it in this document.
https://pypi.org/search/?q=&o=&c=Framework+%3A%3A+Sphinx+%3A%3A+Extension

**Comment by stephenfin** [comment]
> LGTM. BTW, PyPI provides a classifier for sphinx extensions. So it would be nice if we have a reference for it in this document.
> [pypi.org/search/?q=&o=&c=Framework+%3A%3A+Sphinx+%3A%3A+Extension](https://pypi.org/search/?q=&o=&c=Framework+%3A%3A+Sphinx+%3A%3A+Extension)

Sure, I'll do a follow-up for this.

**Comment by stephenfin** [comment]
> > LGTM. BTW, PyPI provides a classifier for sphinx extensions. So it would be nice if we have a reference for it in this document.
> > [pypi.org/search/?q=&o=&c=Framework+%3A%3A+Sphinx+%3A%3A+Extension](https://pypi.org/search/?q=&o=&c=Framework+%3A%3A+Sphinx+%3A%3A+Extension)
> 
> Sure, I'll do a follow-up for this.

Done.

File: doc/usage/extensions/index.rst [hunk 1]
```diff
@@ -41,22 +41,19 @@ These extensions are built in and can be activated by respective entries in the
...
-list (`join here <https://groups.google.com/forum/#!forum/sphinx-dev>`_).
-
-.. _Sphinx Contrib: https://bitbucket.org/birkenfeld/sphinx-contrib
+You can find several extensions contributed by users in the `sphinx-contrib`__
+organization. If you wish to include your extension in this organization,
+simply follow the instructions provided in the `github-administration`__
+project. This is optional and there are several extensions hosted elsewhere.
+The `awesome-sphinxdoc`__ project contains a curated list of Sphinx packages,
+and many packages use the ``Framework :: Sphinx :: Extension`` and
+``Framework :: Sphinx :: Theme`` `trove classifiers`__ for Sphinx extensions
```
Comment (tk0miya) [inline doc/usage/extensions/index.rst:50]: It would be nice if we add a hyperlink to search result pages for these trove classifiers. What do you think?

**Comment by tk0miya** [comment]
Merged. I'll post another PR for the classifiers later.

**Pull request merged**